libcudacxx/include/cuda/__mdspan/host_device_accessor.h

```diff
@@ -0,0 +1,273 @@
+
+template <typename _Accessor>
+struct managed_accessor;
+
+/***********************************************************************************************************************
+ * Host/Device/Managed Accessor Traits
+ **********************************************************************************************************************/
+
+template <typename>
+inline constexpr bool is_host_accessor_v = false;
+
+template <typename>
+inline constexpr bool is_device_accessor_v = false;
+
+template <typename>
+inline constexpr bool is_managed_accessor_v = false;
+
+template <typename _Accessor>
+inline constexpr bool is_host_accessor_v<host_accessor<_Accessor>> = true;
+
+template <typename _Accessor>
+inline constexpr bool is_device_accessor_v<device_accessor<_Accessor>> = true;
+
+template <typename _Accessor>
+inline constexpr bool is_managed_accessor_v<managed_accessor<_Accessor>> = true;
+
+template <typename _Tp>
+inline constexpr bool is_host_device_managed_accessor_v =
+  is_host_accessor_v<_Tp> || is_device_accessor_v<_Tp> || is_managed_accessor_v<_Tp>;
+
+/***********************************************************************************************************************
+ * Host Accessor
+ **********************************************************************************************************************/
+
+template <typename _Accessor>
+struct host_accessor : public _Accessor
+{
+private:
+  using __data_handle_type = typename _Accessor::data_handle_type;
+  using __reference        = typename _Accessor::reference;
+  using __self             = host_accessor<_Accessor>;
+
+  static constexpr bool __is_ctor_noexcept      = noexcept(_Accessor{});
+  static constexpr bool __is_copy_ctor_noexcept = noexcept(_Accessor{});
+  static constexpr bool __is_access_noexcept    = noexcept(_Accessor{}.access(__data_handle_type{}, 0));
+  static constexpr bool __is_offset_noexcept    = noexcept(_Accessor{}.offset(__data_handle_type{}, 0));
+
+  static_assert(!is_host_device_managed_accessor_v<_Accessor>,
+                "cuda::host_accessor/cuda::device_accessor/cuda::managed_accessor cannot be nested");
+
+  template <typename __data_handle_type>
+  _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool __is_host_accessible_pointer(__data_handle_type __p) noexcept
+  {
+    if constexpr (_CUDA_VSTD::is_pointer_v<__data_handle_type>)
+    {
+      cudaPointerAttributes __attrib;
+      _CCCL_VERIFY(::cudaPointerGetAttributes(&__attrib, __p) == ::cudaSuccess, "cudaPointerGetAttributes failed");
+      return __attrib.hostPointer != nullptr || __attrib.type == ::cudaMemoryTypeUnregistered;
+    }
+    else
+    {
+      return true; // cannot be verified
+    }
+  }
+
+  _LIBCUDACXX_HIDE_FROM_ABI static constexpr void __check_host_pointer(__data_handle_type __p) noexcept
+  {
+    _CCCL_ASSERT(__self::__is_host_accessible_pointer(__p), "cuda::host_accessor data handle is not a HOST pointer");
+  }
+
+public:
+  using offset_policy = host_accessor;
```

Comment:
This is not correct, because it assumes that `_Accessor::offset_policy` is just `_Accessor`.

```suggestion
  using offset_policy = host_accessor<typename _Accessor::offset_policy>;
```

Reply:
This remark applies to `device_accessor` and `managed_accessor` as well.